internal/services/instance_types_service_test.go

```diff
@@ -0,0 +1,70 @@
+package services_test
+
+import (
+	"context"
+	"encoding/json"
+	"fmt"
+	"net/http"
+	"net/http/httptest"
+	"testing"
+
+	"github.com/RHEnVision/provisioning-backend/internal/clients"
+	clientStub "github.com/RHEnVision/provisioning-backend/internal/clients/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/dao/stubs"
+	"github.com/RHEnVision/provisioning-backend/internal/payloads"
+	"github.com/RHEnVision/provisioning-backend/internal/services"
+	"github.com/RHEnVision/provisioning-backend/internal/testing/identity"
+	"github.com/stretchr/testify/assert"
+)
+
+func TestListInstanceTypes(t *testing.T) {
+
+	t.Run("with region", func(t *testing.T) {
+		var names []string
+		ctx := stubs.WithAccountDaoOne(context.Background())
+		ctx = identity.WithTenant(t, ctx)
+		ctx = clientStub.WithSourcesClient(ctx)
+		ctx = clientStub.WithEC2Client(ctx)
+
+		req, err := http.NewRequestWithContext(ctx, "GET", "/api/provisioning/sources/1/instance_types?region=us-east-1", nil)
+		assert.Nil(t, err, fmt.Sprintf("Error creating a new request: %v", err))
+
+		rr := httptest.NewRecorder()
+		handler := http.HandlerFunc(services.ListInstanceTypes)
+		handler.ServeHTTP(rr, req)
+
+		if status := rr.Code; status != http.StatusOK {
+			t.Errorf("Handler returned wrong status code. Expected: %d. Got: %d.", http.StatusOK, status)
+		}
+
+		var result []clients.InstanceType
+
+		if err := json.NewDecoder(rr.Body).Decode(&result); err != nil {
+			t.Errorf("Error decoding response body: %v", err)
+		}
+
+		assert.Equal(t, 3, len(result), "expected three result in response json")
+		for _, it := range result {
+			names = append(names, it.Name)
+		}
+		assert.Contains(t, names, "a1.2xlarge", "expected result to contain a1.2xlarge instance type")
+		assert.Contains(t, names, "c5.xlarge", "expected result to contain c5.xlarge instance type")
```

Comment:
Like here, this is too much typing since you can tell from the error message Go framework generates. It literally says "this value should be X but it was Y, on this line" without any message:

```suggestion
		assert.Contains(t, names, "c5.xlarge")
```

But hey, if you really like to be verbose, so be it.